### PR TITLE
feat: 🎸 Import/Export grammar + trrack update

### LINKED ISSUE #101: Remove trrackjs peer dependencies when packaging is updated

### PR DESCRIPTION
**Pull request by JakeWags**
### Does this PR close any open issues?
Closes #101 #108 

### Give a longer description of what this PR addresses and why it's needed
Adds importing and exporting of an Upset state in the form of a JSON upset grammar.
Also, updates Trrack to Trrack 1.0.0 and removes peer dependencies which are now bundled with Trrack

### CODE REVIEW

**Comment by JakeWags** [comment]
Needs Trrack update to 1.0.1

File: packages/upset/src/provenance/index.ts [hunk 1]
```diff
@@ -179,6 +179,12 @@ const removePlotAction = registry.register('remove-plot',
   },
 );
 
+const replaceStateAction = registry.register('set-state',
+  (state: UpsetConfig, newState: UpsetConfig) => {
+    console.log("State change from", JSON.stringify(state), "to", JSON.stringify(newState));
```
Comment (JackWilb) [inline packages/upset/src/provenance/index.ts:184]: Should we remove this console.log?

File: packages/upset/src/components/Sidebar.tsx [hunk 2]
```diff
@@ -53,12 +55,18 @@ export const Sidebar = () => {
     secondAggregateBy !== 'None',
   );
 
+  const [ showImportModal, setShowImportModal] = useState(false);
```
Comment (JackWilb) [inline packages/upset/src/components/Sidebar.tsx:58]:
```suggestion
  const [ showImportModal, setShowImportModal ] = useState(false);
```

We need to add a linter for this project

File: packages/upset/src/components/ImportModal.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+export const ImportModal = (props:{open: boolean, close: () => void}) => {
+  const [ isError, setIsError ] = useState(false);
+  const { actions } = useContext(
+    ProvenanceContext,
+  );
+
+  async function readFile(file: File) {
+    const data = await file.text();
+  
+    const newState = JSON.parse(data);
```
Comment (JackWilb) [inline packages/upset/src/components/ImportModal.tsx:14]: We will eventually need this to be HJSON compatible. We should make an issue or just fix it now. 1 library and a changed call here

File: packages/upset/src/components/ImportModal.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                  input !== null && input.files !== null && 
+                  readFile(input.files[0])
+                  .catch(() => setIsError(true))
+                }}>
+                  Submit
+              </Button>
+        </DialogContent>
+          <Snackbar
+            open={isError} autoHideDuration={6000} onClose={() => setIsError(false)}>
+            <Alert severity="error">Incompatible File Type! Please import a .json Upset State file.</Alert>
```
Comment (JackWilb) [inline packages/upset/src/components/ImportModal.tsx:42]: json/hjson

File: packages/upset/src/components/ImportModal.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+          </Snackbar>
+    </Dialog>
+  );
+};
+
+export const exportStateGrammar = (provenance: any) => {
+    const fileName = "upset_state";
```
Comment (JackWilb) [inline packages/upset/src/components/ImportModal.tsx:49]: Should we add a datetime to this?

**Review by JackWilb — APPROVED** [review]
A couple small suggestions

File: packages/upset/src/provenance/index.ts (hunk 1 above)
Comment (kirangadhave) [reply packages/upset/src/provenance/index.ts:184]: We should

File: packages/upset/src/components/Sidebar.tsx (hunk 2 above)
Comment (kirangadhave) [reply packages/upset/src/components/Sidebar.tsx:58]: I thought I had a linter setup for this. We should add a precommit hook to lint files like I do for Trrack. Right now linting relies on having ide extension

File: packages/upset/src/components/ImportModal.tsx (hunk 5 above)
Comment (kirangadhave) [reply packages/upset/src/components/ImportModal.tsx:49]: Wouldn’t hurt, but it should be optional

Comment (JakeWags) [reply packages/upset/src/components/ImportModal.tsx]: I added this as yyyy-mm-dd format. total file name is `upset_state_2023-02-15.json` for example

**Pull request merged**